pkg/manager/integration/manager_test.go

```diff
@@ -0,0 +1,286 @@
+	//   * Set up a Driver reconciler to verify reconciliation works
+	//   * Set up a conversion webhook which only works if readiness probe succeeds (just like a Kubernetes service)
+	//   * Add an index on v2 Driver to ensure we start and wait for an informer during cache.Start (as part of manager.Start)
+	//     * Note: cache.Start would fail if the conversion webhook doesn't work (which in turn depends on the readiness probe)
+	Describe("Start should start all components without deadlock", func() {
+		ctx := ctrl.SetupSignalHandler()
+
+		// Set up schema.
+		Expect(clientgoscheme.AddToScheme(scheme)).To(Succeed())
+		Expect(apiextensionsv1.AddToScheme(scheme)).To(Succeed())
+		Expect(crewv1.AddToScheme(scheme)).To(Succeed())
+		Expect(crewv2.AddToScheme(scheme)).To(Succeed())
+
+		// Set up test environment.
+		env := &envtest.Environment{
+			Scheme: scheme,
+			CRDInstallOptions: envtest.CRDInstallOptions{
+				CRDs: []*apiextensionsv1.CustomResourceDefinition{driverCRD},
+			},
+		}
+		cfg, err := env.Start()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(cfg).NotTo(BeNil())
+		defer func() {
+			Expect(env.Stop()).To(Succeed())
+		}()
+		c, err := client.New(cfg, client.Options{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Create driver CR (which is stored as v1).
+		driverV1 := &unstructured.Unstructured{}
+		driverV1.SetGroupVersionKind(crewv1.GroupVersion.WithKind("Driver"))
+		driverV1.SetName("driver1")
+		driverV1.SetNamespace(metav1.NamespaceDefault)
+		Expect(c.Create(ctx, driverV1)).To(Succeed())
+
+		// Update driver CRD to make v2 the storage version.
+		driverCRDV2Storage := driverCRD.DeepCopy()
+		driverCRDV2Storage.Spec.Versions[0].Storage = false
+		driverCRDV2Storage.Spec.Versions[0].Storage = true
+		Expect(c.Patch(ctx, driverCRDV2Storage, client.MergeFrom(driverCRD))).To(Succeed())
+
+		// Set up Manager.
+		ctrl.SetLogger(zap.New())
+		mgr, err := manager.New(env.Config, manager.Options{
+			Scheme:                 scheme,
+			HealthProbeBindAddress: ":0",
+			// Disable metrics to avoid port conflicts.
+			MetricsBindAddress: "0",
+			WebhookServer: webhook.NewServer(webhook.Options{
+				Port:    env.WebhookInstallOptions.LocalServingPort,
+				Host:    env.WebhookInstallOptions.LocalServingHost,
+				CertDir: env.WebhookInstallOptions.LocalServingCertDir,
+			}),
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Configure health probes.
+		Expect(mgr.AddReadyzCheck("webhook", mgr.GetWebhookServer().StartedChecker())).To(Succeed())
+		Expect(mgr.AddHealthzCheck("webhook", mgr.GetWebhookServer().StartedChecker())).To(Succeed())
+
+		// Set up Driver reconciler.
+		driverReconciler := &DriverReconciler{
+			Client: mgr.GetClient(),
+		}
+		Expect(ctrl.NewControllerManagedBy(mgr).For(&crewv1.Driver{}).Complete(driverReconciler)).To(Succeed())
+
+		// Set up a conversion webhook.
+		conversionWebhook := createConversionWebhook(mgr)
+		mgr.GetWebhookServer().Register("/convert", conversionWebhook)
```

Comment:
I might have missed this: where are we registering this webhook in the CRD itself so the apiserver knows where to go?

Reply:
Auto-magically done by testenv on startup when deploying the CRD (it does this for all types that are convertible)

I'll add a godoc comment